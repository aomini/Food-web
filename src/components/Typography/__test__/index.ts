import { render } from '@testing-library/react';

import Typography from '..';

describe('<Typography />', () => {
  it('should correctly render the typography with variants', () => {});

  it('should render the typography element tag properly', () => {});
});
