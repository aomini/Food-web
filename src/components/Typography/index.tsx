import { ElementType, HtmlHTMLAttributes } from 'react';

/**
 * Variants Mapping
 * @author Rakesh Shrestha
 */
export const variantMapping = {
  h1: {
    style: 'text-4xl',
    mapping: 'h1'
  },
  h2: {
    style: 'text-3xl',
    mapping: 'h2'
  },
  h3: {
    style: 'text-2xl',
    mapping: 'h3'
  },
  h4: {
    style: 'text-xl',
    mapping: 'h4'
  },
  h5: {
    style: 'text-lg',
    mapping: 'h5'
  },
  body: {
    style: 'text-base',
    mapping: 'p'
  },
  caption: {
    style: 'text-sm',
    mapping: 'span'
  }
};

type TypographyTypes = {
  variant: keyof typeof variantMapping;
  component?: ElementType;
};

const Typography: React.FC<
  TypographyTypes & HtmlHTMLAttributes<HTMLParagraphElement>
> = ({ variant, component, children }) => {
  const style = variantMapping[variant].style;

  const Component = component || variantMapping[variant].mapping || 'span';

  return <Component className={style}>{children}</Component>;
};
export default Typography;
