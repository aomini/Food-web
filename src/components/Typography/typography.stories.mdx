import { Meta, Typeset } from '@storybook/addon-docs/blocks'
import Typography from '.'

<Meta title="Typography/Base" />

export const typography = {
  type: {
    primary: '"Open Sans", sans-serif'
  },
  weight: {
    regular: '400',
    semibold: '600',
    bold: '700'
  },
  size: {
    xs: 12,
    sm: 14,
    base: 16,
    lg: 18,
    xl: 20,
    '2xl': 28,
    '3xl': 32,
    '4xl': 60
  }
}

export const SampleText = 'Rakesh Shrestha is the best react developer around'

# Base Typography

**Font:** Open Sans

**Weights:** 400(regular), 500(bold), 800(extrabold), 900(black)

<Typeset
  fontSizes={[
    Number(typography.size.xs),
    Number(typography.size.sm),
    Number(typography.size.base),
    Number(typography.size.lg),
    Number(typography.size.xl),
    Number(typography.size['2xl']),
    Number(typography.size['3xl']),
    Number(typography.size['4xl'])
  ]}
  fontWeight={typography.weight.bold}
  sampleText={SampleText}
  fontFamily={typography.type.primary}
/>
