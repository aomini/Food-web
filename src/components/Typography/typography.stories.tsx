import Typography from '.';

export default {
  title: 'Typography/styles'
};

export const Header = () => (
  <>
    <Typography variant="h1">Headers/h1</Typography>
    <Typography variant="h2">Headers/h2</Typography>
    <Typography variant="h3">Headers/h3</Typography>
    <Typography variant="h4">Headers/h4</Typography>
    <Typography variant="h5" component="h5">
      Headers/h5
    </Typography>
  </>
);

export const TextBase = () => (
  <>
    <Typography variant="body">Headers/body</Typography>
    <Typography variant="caption" component="span">
      Headers/caption
    </Typography>
  </>
);
