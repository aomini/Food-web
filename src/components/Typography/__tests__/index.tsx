import { render } from '@testing-library/react';

import Typography, { variantMapping } from '..';

describe('<Typography />', () => {
  it('should correctly render with the given variant', () => {
    const { container, getByRole } = render(
      <Typography variant="h2">Heading 2</Typography>
    );

    expect(getByRole('heading')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a typography element with mapped variant', () => {
    const mappedVariants: Array<keyof typeof variantMapping> = [
      'body',
      'caption'
    ];

    mappedVariants.forEach((variant) => {
      const { container } = render(
        <Typography variant={variant}>Paragraph</Typography>
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should render a typography element with supplied variant', () => {
    const { container, getByRole } = render(
      <Typography variant="body" component="h6">
        Text rendered with heading
      </Typography>
    );

    expect(
      getByRole('heading', { name: /Text rendered with heading/i })
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
