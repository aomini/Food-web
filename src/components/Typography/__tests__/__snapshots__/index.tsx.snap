// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> should correctly render with the given variant 1`] = `
<h2
  class="text-3xl"
>
  Heading 2
</h2>
`;

exports[`<Typography /> should render a typography element with mapped variant 1`] = `
<p
  class="text-base"
>
  Paragraph
</p>
`;

exports[`<Typography /> should render a typography element with mapped variant 2`] = `
<span
  class="text-sm"
>
  Paragraph
</span>
`;

exports[`<Typography /> should render a typography element with supplied variant 1`] = `
<h6
  class="text-base"
>
  Text rendered with heading
</h6>
`;
